include/common/bitdepth.rs

```diff
@@ -193,6 +204,34 @@ pub trait BitDepth: Clone + Copy {
 
     fn get_intermediate_bits(&self) -> u8;
 
+    fn cast_pixel_slice(bytes: &[u8]) -> &[Self::Pixel] {
+        let size = mem::size_of::<Self::Pixel>();
+
+        // Check that the number of elements is a multiple of the new element
+        // size and that the alignment is correct for the new element type.
+        assert!(bytes.len() % size == 0);
```

Comment:
```suggestion
        debug_assert!(bytes.len() % size == 0);
```

Not being a multiple shouldn't happen, but it's not UB if it does, so this can just be a `debug_assert!`.  If we use `BPC::pxstride` from #780`, it does that same `debug_assert!` with `%`.  Up to you if you think we should still just inline it here, which does make the safety much clearer.

Same for the `mut` version, too, of course.

Reply:
I'm going to leave the assert inlined, but I'll change it to `debug_assert!`. I had initially been thinking of the case where the source buffer has an odd number of elements, so converting to a `&[u16]` buffer would have a half element at the end. But since we're doing integer division in that case the new length will always be truncated, so we would actually omit that last byte entirely. Though in practice none of this should matter because the buffers are created with the correct length 🤷